File: test/end-to-end/Classes/declarations.cs
//
class Point
{
    public int X;
    public int Y;
    protected int Square;
    private string name;

    public Point(int x, int y)
    {
        this.X = x;
        this.Y = y;
        this.Square = x * y;
        this.name = "";
    }
}

class Measurements<T>
{
    public Measurements(T[] items)
    {
        // ...
    }
}

class Shape : Measurements<Point>
{
    public Shape(Point[] points)
    {
        // ...
    }
}
//

File: test/end-to-end/Classes/declarations.ts
//
class Point {
    public x: number;
    public y: number;
    protected square: number;
    private name: string;

    public constructor(x: number, y: number) {
        this.x = x;
        this.y = y;
        this.square = x * y;
        this.name = "";
    }
}

class Measurements<T> {
    public constructor(items: T[]) {
        // ...
    }
}

class Shape extends Measurements<Point> {
    public constructor(points: Point[]) {
        // ...
    }
}
//
